File: spring-boot-java/src/main/java/com/example/springbootjava/repository/MovieRepository.java
package com.example.springbootjava.repository;

import com.example.springbootjava.document.Movie;
import org.springframework.data.mongodb.repository.Aggregation;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.List;

public interface MovieRepository extends MongoRepository<Movie, String> {

    @Aggregation("{ $sample: {size: 1} }")
    Movie findOneRandomMovie();
    @Aggregation("{ $sample: {size: ?0} }")
    List<Movie> findRandomMovies(Integer count);
}

File: spring-webflux-kotlin/src/main/kotlin/com/example/springwebfluxkotlin/repository/MovieRepository.kt
package com.example.springwebfluxkotlin.repository

import com.example.springwebfluxkotlin.document.Movie
import kotlinx.coroutines.flow.Flow
import org.springframework.data.mongodb.repository.Aggregation
import org.springframework.data.repository.kotlin.CoroutineCrudRepository

interface MovieRepository : CoroutineCrudRepository<Movie, String> {

    @Aggregation("{ \$sample: {size: 1} }")
    fun findOneRandomMovie(): Flow<Movie>

    @Aggregation("{ \$sample: {size: ?0} }")
    fun findRandomMovies(count: Int): Flow<Movie>
}